src/runtime/entitlements-manager-test.js

```diff
@@ -2172,6 +2172,82 @@ describes.realWin('EntitlementsManager', {}, (env) => {
       expect(entitlements.getEntitlementForThis().source).to.equal('google');
       expect(toastOpenStub).to.not.be.called;
     });
+
+    describe('enableDefaultMeteringHandler', () => {
+      beforeEach(() => {
+        expectGetIsReadyToPayToBeCalled(null);
+        expectGetSwgUserTokenToBeCalled();
+        storageMock
+          .expects('set')
+          .withExactArgs(Constants.USER_TOKEN, 'abc', true);
+        jwtHelperMock
+          .expects('decode')
+          .withExactArgs('SIGNED_DATA')
+          .returns({
+            entitlements: {
+              source: GOOGLE_METERING_SOURCE,
+              products: ['pub1:label1'],
+              subscriptionToken: 'token1',
+            },
+          });
+        jwtHelperMock
+          .expects('decode')
+          .withExactArgs('token1')
+          .returns({
+            metering: {
+              ownerId: 'scenic-2017.appspot.com',
+              action: 'READ',
+              clientUserAttribute: 'standard_registered_user',
+              clientType: MeterClientTypes.METERED_BY_GOOGLE.valueOf(),
+            },
+          });
+        xhrMock
+          .expects('fetch')
+          .withExactArgs(
+            '$frontend$/swg/_/api/v1/publication/pub1/entitlements',
+            {
+              method: 'GET',
+              headers: {'Accept': 'text/plain, application/json'},
+              credentials: 'include',
+            }
+          )
+          .returns(
+            Promise.resolve({
+              text: () =>
+                Promise.resolve(
+                  JSON.stringify({
+                    signedEntitlements: 'SIGNED_DATA',
+                    swgUserToken: 'abc',
+                  })
+                ),
+            })
+          );
+      });
+
+      it('should NOT consume a Google meter entitlement by default', async () => {
+        const entitlements = await manager.getEntitlements();
```

Comment:
I definitely agree that the setup (even assertions included by default) are not clear due to the default before/after functions. We could just ignore the default setups and make our own - we should be pretty diligent at enforcing that pattern if we go down that route since there is already so much test code following it.

The assertions are (unfortunately) just included in the default afterAlls - one option could just be explicitly checking the mocks that have expected functionality we are explicitly setting up. Let me switch to explicitly checking instead of implicit.

Reply:
I know there are assertions in afterAll blocks checking that all the mocks are called with expected arguments, and agree this is an unfortunate pattern. Assertions should also be moved out of before blocks wherever possible. In this case, the test isn't adding any specific mocks with expected arguments and the only specific assertion is the same as in the test for the other case on L2248, so I'm not seeing what this is actually testing.

Reply:
Moved the setup/expectation code into each function respectively. Commented that the specific check that existed here was to assert that the handler code _should_ have triggered if it were enabled